### PR TITLE
refactor: update promise handle

### PR DESCRIPTION
**Pull request by xzf0587**

### CODE REVIEW

File: packages/api/src/utils/login.ts [hunk 1]
```diff
@@ -231,9 +231,9 @@ export abstract class BasicLogin {
     return ok(true);
   }
 
-  async removeStatusChangeMap(name: string): Promise<Result<boolean, FxError>> {
+  removeStatusChangeMap(name: string): Promise<Result<boolean, FxError>> {
     this.statusChangeMap.delete(name);
-    return ok(true);
+    return Promise.resolve(ok(true));
```
Comment (blackchoey) [inline packages/api/src/utils/login.ts:236]: What's the effort if we also update the return type from `Promise<Result<boolean, FxError>>` to `Result<boolean, FxError>`? Seems there's no need to return a promise here.

Comment (xzf0587) [reply packages/api/src/utils/login.ts:236]: This method is in the M365TokenProvider interface.
There are concrete implementations in vscode/cli.
We need to change them to keep the method interface consistant.

**Review by SLdragon — APPROVED** [review]

**Review by yukun-dong — APPROVED** [review]

**Review by blackchoey — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

File: packages/fx-core/src/core/middleware/utils/v3MigrationUtils.ts [hunk 2]
```diff
@@ -278,7 +278,7 @@ export async function readAndConvertUserdata(
 ): Promise<string> {
   let returnAnswer = "";
 
-  const userdataContent = fs.readFileSync(path.join(context.projectPath, filePath), "utf8");
+  const userdataContent = await fs.readFile(path.join(context.projectPath, filePath), "utf8");
```
Comment (frankqianms) [inline packages/fx-core/src/core/middleware/utils/v3MigrationUtils.ts:281]: The same question as Chaoyi's: why we need to make this change, they look the same

Comment (xzf0587) [reply packages/fx-core/src/core/middleware/utils/v3MigrationUtils.ts:281]: It is ok to update this function to sync function. 
If so, we need to handle the thrown out error of readFile which the caller of the function should update as well.
If we use await inside the function, the caller of the function has no need to change.

**Pull request merged**